SQL/proc_descriptions/procedure_assign_average_speed_to_all_segments_in_area.md

```diff
@@ -0,0 +1,110 @@
+# Assign average speed to all segments in area Decsription
+## Purpose of this file
+The purpose of this file is to ensure that there are as little misunderstandings as could be. That's why it serves to show how I (Vladyslav Zlochevskyi/zlochina) see what this procedure does. Also this way I might figure out what actually could be tested in this procedure.
+
+## Decsription of the procedure
+- Procedure name: procedure_assign_average_speed_to_all_segments_in_area
+- The __name__ of the procedure + __arguments__ of the procedure implicate computing average speed per segment of an area and adding this speed info to corresponding tables.
+- Input params: `target_area_id:smallint`, `target_area_srid:integer`. Return value: _None_.
+- Target of the procedure: gather some info and insert it into table `nodes_ways_speeds`
+- Flow:
+    - Prepare for final insertion to `nodes_ways_speeds`, by executing CTE such as `average_speed`, `target_ways` and `node_segments`:
+        - CTE `average_speed` aggregates all records from `nodes_ways_speeds` table to one record containing average speed and standard deviation filtered by quality either equal to 1 or 2.
+        - CTE `target_ways` executes function `get_ways_in_target_area`, which returns table used by `node_segments` CTE. Note: Did not figure out the role it demonstrates to the procedure.
```

Comment:
Speeds are assigned to ways. That's why you need to get the ways for assignment first

Reply:
Well, actually, it seems like source of ways are coming not from this CTE, but rather from table `nodes_ways` in `node_segments` CTE. So far, this `target_ways` is used only for filtering out some records, which are not included both in `nodes_ways` and the table returned by `get_ways_in_target_area()` function (inclusion by "nodes_ways.way_id and ways.id")

Reply:
well that's possible that some tables are used only for filtration